packages/ddp-client/client/queueStubsHelpers.js

```diff
@@ -0,0 +1,226 @@
+            );
+          }
+        });
+      },
+      {
+        stubPromise,
+        serverPromise,
+      }
+    );
+  };
+
+  let oldApply = Meteor.connection.apply;
+  Meteor.connection.apply = function () {
+    // [name, args, options]
+    let options = arguments[2] || {};
+    let wait = options.wait;
+
+    // Apply runs the stub before synchronously returning.
+    //
+    // However, we want the server to run the methods in the original call order
+    // so we have to queue sending the message to the server until any previous async
+    // methods run.
+    // This does mean the stubs run in a different order than the methods on the
+    // server.
+    // TODO: can we queue Meteor.apply in some situations instead of running
+    // immediately?
+
+    let oldOutstandingMethodBlocks = Meteor.connection._outstandingMethodBlocks;
+    // Meteor only sends the method if _outstandingMethodBlocks.length is 1.
+    // Add a wait block to force Meteor to put the new method in a second block.
+    let outstandingMethodBlocks = [{ wait: true, methods: [] }];
+    Meteor.connection._outstandingMethodBlocks = outstandingMethodBlocks;
+
+    let result;
+    try {
+      result = oldApply.apply(this, arguments);
+    } finally {
+      Meteor.connection._outstandingMethodBlocks = oldOutstandingMethodBlocks;
+    }
+
+    if (outstandingMethodBlocks[1]) {
+      let methodInvoker = outstandingMethodBlocks[1].methods[0];
+
+      if (methodInvoker) {
+        queueMethodInvoker(methodInvoker, wait);
+      }
+    }
+
+    return result;
+  };
+
+  function queueMethodInvoker(methodInvoker, wait) {
+    queueFunction((resolve) => {
+      let self = Meteor.connection;
+      // based on https://github.com/meteor/meteor/blob/e0631738f2a8a914d8a50b1060e8f40cb0873680/packages/ddp-client/common/livedata_connection.js#L833-L853C1
+      if (wait) {
+        // It's a wait method! Wait methods go in their own block.
+        self._outstandingMethodBlocks.push({
+          wait: true,
+          methods: [methodInvoker],
+        });
+      } else {
+        // Not a wait method. Start a new block if the previous block was a wait
+        // block, and add it to the last block of methods.
+        if (
+          isEmpty(self._outstandingMethodBlocks) ||
+          last(self._outstandingMethodBlocks).wait
+        ) {
+          self._outstandingMethodBlocks.push({
+            wait: false,
+            methods: [],
+          });
+        }
+
+        last(self._outstandingMethodBlocks).methods.push(methodInvoker);
+      }
+
+      // If we added it to the first block, send it out now.
+      if (self._outstandingMethodBlocks.length === 1)
+        methodInvoker.sendMessage();
+
+      resolve();
+    });
+  }
+
+  /**
+   * Queue subscriptions in case they rely on previous method calls
+   */
+  let queueSend = false;
+  let oldSubscribe = Meteor.connection.subscribe;
+  Meteor.connection.subscribe = function () {
```

Comment:
A test for the Meteor.subscribe wrapper is missing (there was one in the original code).

Reply:
Couldn't find this test, can you send it here, please?

Reply:
Sorry, I forgot to push it to github. Here is the commit: https://github.com/zodern/fix-async-stubs/commit/fd6519351f6f0c386cd5ec18d43b26c1aadf7678